things/mocks/channels.go

```diff
@@ -78,30 +78,40 @@ func (crm *channelRepositoryMock) RetrieveByID(_ context.Context, owner, id stri
 }
 
 func (crm *channelRepositoryMock) RetrieveAll(_ context.Context, owner string, pm things.PageMetadata) (things.ChannelsPage, error) {
-	if pm.Limit <= 0 {
+	if pm.Limit < 0 {
```

Comment:
It's 0 limit valid?

Reply:
yes, limit 0 would not throw error, and in that case you will get default limi (10)